include/exadg/functions_and_boundary_conditions/interface_coupling.h

```diff
@@ -145,13 +109,101 @@ class InterfaceCoupling
       }
 
       array_solution_dst.resize(array_q_points_dst.size(), dealii::Tensor<rank, dim, double>());
+    }
+
+    // finally, give boundary condition access to the data
+    for(auto boundary : map_bc_)
+    {
+      boundary.second->set_data_pointer(map_vector_index, map_solution);
+    }
+  }
+
+  std::vector<quad_index> const &
+  get_quad_indices()
+  {
+    return quad_indices;
+  }
+
+  ArrayQuadraturePoints &
+  get_array_q_points(quad_index const & q_index)
+  {
+    return map_q_points[q_index];
+  }
+
+  ArraySolutionValues &
+  get_array_solution(quad_index const & q_index)
+  {
+    return map_solution[q_index];
+  }
 
-      /*
-       * 2. Communication: exchange quadarature points with their owners
-       */
-      map_evaluator.emplace(quadrature,
+private:
+  std::vector<quad_index> quad_indices;
+
+  mutable std::map<quad_index, MapVectorIndex>        map_vector_index;
+  mutable std::map<quad_index, ArrayQuadraturePoints> map_q_points;
+  mutable std::map<quad_index, ArraySolutionValues>   map_solution;
+};
+
+template<int dim, int n_components, typename Number>
+class InterfaceCoupling
+{
+private:
+  static unsigned int const rank =
+    (n_components == 1) ? 0 : ((n_components == dim) ? 1 : dealii::numbers::invalid_unsigned_int);
+
+  using MapBoundaryCondition =
+    std::map<dealii::types::boundary_id, std::shared_ptr<FunctionCached<rank, dim, double>>>;
```

Comment:
Since we are here. I think we should not compute `rank` ourselves and instead use:

https://github.com/dealii/dealii/blob/2105cad54b490002bf27ec6cc26d326c8eb057d0/include/deal.II/numerics/vector_tools_evaluate.h#L116-L119

Using `Tensor` here will not work if we use these classes for scalar valued solutions.

Reply:
I consider this also a follow-up PR. I currently don't know how to deal with the value_type/double mixture that we have in ExaDG, due to compatibility with dealii-9.3 on the one hand and due to the fact that we do not want to introduce a second template parameter `Number` for `BoundaryDescriptor`s and `Function`s  on the other hand (even though I am open to changing this in ExaDG in the future ... but I think we should discuss this in a broader context, e.g. as separate issues in ExaDG).